content/en/blog/2023/spring-native/index.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Observe your Spring Native image application with OpenTelemetry
+linkTitle: Spring Native
+date: 2023-11-28
+author: >-
+  [Jean Bisutti](https://github.com/jeanbisutti) (Microsoft)
+cSpell:ignore: datasource logback bisutti
+---
+
+You may know the
+[OpenTelemetry Java agent](https://opentelemetry.io/docs/instrumentation/java/automatic/).
+However, Java agents are
+[not officially supported](https://github.com/oracle/graal/issues/1065) with
+GraalVM native images today. This post presents an OpenTelemetry solution to get
+telemetry data with Spring Boot native image applications.
+
+## A history of the last months
+
+The OpenTelemetry contributors have used the existing OpenTelemetry Spring Boot
+starter to instrument Spring Boot native image applications with OpenTelemetry
+without byte code instrumentation. They have improved the starter in the last
+months. For example, they have added the OTLP log export. The OpenTelemetry
+starter provides auto-instrumentation for Spring HTTP frameworks. So, by adding
+the starter dependency to your project, you will benefit from the
+instrumentation of HTTP requests:
+
+```xml
+<dependencies>
+  <dependency>
+    <groupId>io.opentelemetry.instrumentation</groupId>
+    <artifactId>opentelemetry-spring-boot-starter</artifactId>
+    <version>1.32.0-alpha</version>
+  </dependency>
+</dependencies>
+```
+
+In addition to the starter,
+[OpenTelemetry instrumentation libraries](https://opentelemetry.io/docs/instrumentation/java/manual/)
+can be used. The OpenTelemetry contributors have improved the JDBC (database)
+libraries and logging instrumentation libraries. For the Logback logging
+library, the OpenTelemetry developers have added GraalVM configuration to make
+the library work in native mode.
+[Spring Boot native image applications do not support Log4j2 logging library](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-with-GraalVM).
+They have also worked to reduce the configuration to set up the logging and
+database instrumentation with the OpenTelemetry Starter. For example, if your
+application does not declare DataSource bean, you enable the database by simply
+adding two properties in the application.properties file:
+
+```properties
+spring.datasource.url=jdbc:otel:h2:mem:db
+spring.datasource.driver-class-name=io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver
+```
```

Comment:
~~~suggestion
To get even more visibility, the Starter can be combined with [instrumentation libraries](/docs/instrumentation/java/libraries/). For this purpose, the OpenTelemetry Java contributors have improved the JDBC (database)
libraries and logging instrumentation libraries. For example, for the Logback logging library, they have added GraalVM configuration to make the library work in native mode[^1].

Furthermore, they have worked to reduce the configuration to set up the logging and
database instrumentation with the Starter. For example, if your
application does not declare a DataSource bean, you can enable the database by simply
adding two properties in the `application.properties` file:

```properties
spring.datasource.url=jdbc:otel:h2:mem:db
spring.datasource.driver-class-name=io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver
```
~~~